src/PotentialSimulation/ConvergenceAndRefinement.jl

```diff
@@ -88,282 +88,181 @@ function _update_till_convergence!( fssrb::PotentialSimulationSetupRB{T, N1, N2}
+_get_closed_ticks(ticks::Vector{T}, int::Interval{:open, :closed, T}) where {T} = vcat([int.left], ticks)
 
-function _get_refinement_inds(  potential::Array{T, 3}, grid::Grid{T, 3, Cartesian}, 
-                                max_diffs::NTuple{3, T}, minimum_distances::NTuple{3, T})::NTuple{3, Vector{Int}} where {T}
-    inds_x::Vector{Int} = Int[]
-    inds_y::Vector{Int} = Int[]
-    inds_z::Vector{Int} = Int[]
-
-    ax1::DiscreteAxis{T} = grid[1]
-    ax2::DiscreteAxis{T} = grid[2]
-    ax3::DiscreteAxis{T} = grid[3]
-    int1::Interval = ax1.interval
-    int2::Interval = ax2.interval
-    int3::Interval = ax3.interval
+_get_closed_values(values::Array{T,3}, dim::Int, int::ClosedInterval{T}) where {T} = values
 
-    refinement_value_x::T = max_diffs[1]
-    refinement_value_y::T = max_diffs[2]
-    refinement_value_z::T = max_diffs[3]
+function _get_closed_values(values::Array{T,3}, dim::Int, int::Interval{:closed, :open, T})::Array{T,3} where {T} 
+    if dim == 1
+        cat(values, values[1:1,:,:], dims=dim)
+    elseif dim == 2
+        cat(values, values[:,1:1,:], dims=dim)
+    else # dim == 3
+        cat(values, values[:,:,1:1], dims=dim)
+    end
+end
+function _get_closed_values(values::Array{T,3}, dim::Int, int::Interval{:open, :closed, T})::Array{T,3} where {T} 
+    if dim == 1
+        cat(values[end:end,:,:], values, dims=dim)
+    elseif dim == 2
+        cat(values[:,end:end,:], values, dims=dim)
+    else # dim == 3
+        cat(values[:,:,end:end], values, dims=dim)
+    end
+end
 
-    minimum_distance_x::T = minimum_distances[1]
-    minimum_distance_y::T = minimum_distances[2]
-    minimum_distance_z::T = minimum_distances[3]
+function _get_closed_axis(ax::DiscreteAxis{T, BL, BR}) where {T, BL, BR}
+    ticks = _get_closed_ticks(ax.ticks, ax.interval)
+    return DiscreteAxis{T, BL, BR, ClosedInterval{T}}(ClosedInterval(ax.interval.left, ax.interval.right), ticks)
+end
 
-    ax_x::Vector{T} = collect(grid.axes[1])
-    ax_y::Vector{T} = collect(grid.axes[2])
-    ax_z::Vector{T} = collect(grid.axes[3])
+"""
+    _get_closed_potential(p::ScalarPotential{T,3,CS}) where {T, CS}
+
+Returnes an closed Grid & Potential:
+E.g. if one of the axis is {:closed,:open} it will turn this into {:closed,:closed}
+and also extend the `data` field of the potential in the respective dimension and fill
+it with the respective values. 
+"""
+function _get_closed_potential(p::ScalarPotential{T,3,CS}) where {T, CS}
+    g = p.grid
+    closed_values = _get_closed_values(p.data, 1, g.axes[1].interval)
+    closed_values = _get_closed_values(closed_values, 2, g.axes[2].interval)
+    closed_values = _get_closed_values(closed_values, 3, g.axes[3].interval)
+    closed_axes = broadcast(i -> _get_closed_axis(g.axes[i]), (1, 2, 3))
+    AT = typeof(closed_axes)
+    closed_grid = Grid{T, 3, CS, AT}(closed_axes)
+    ElectricPotential{T, 3, CS, AT}(closed_values, closed_grid) 

```

Comment:
Same here: this is by default `ElectricPotential` rather than the initial type of `p`?